### ISSUE TITLE
"unable to parse invalid" message seen when upgrading 1.1.x -> 1.3.3

### ISSUE THREAD

**Issue by timcharper** [mesosphere/marathon#4599]
I'm seeing errors upgrading to Marathon 1.3.3 from Marathon 1.1.x. Looks like a data migration case isn't handled properly.

Right here, we expect that `marathonTaskStatus` exists:

https://github.com/mesosphere/marathon/blob/v1.3.3/src/main/scala/mesosphere/marathon/core/task/tracker/impl/TaskSerializer.scala#L50

However, that field didn't exist in Marathon `1.1.x`. If I call `getMarathonTaskStatus` on a `1.1.x` proto buf task, I get `Invalid`

```@ task.getMarathonTaskStatus
res68: MarathonTask.MarathonTaskStatus = Invalid
```

which is status ID `0` (null object, null value)

```@ res68.getNumber
res70: Int = 0
```

I have opened a WIP merge request which helped migrate the data. It can be a lot more sophisticated but it worked for my case. There are a lot of cases it doesn't handle, unfortunately, and I am not sure how to best handle those.

https://github.com/mesosphere/marathon/pull/4598

**Comment by timcharper** [comment]
Sorry that this isn't in Phabricator. I don't appear to have access. https://phabricator.mesosphere.com/project/view/1/

**Comment by jasongilanfarr** [comment]
Should be fixed in 1.3.4
